### PR TITLE
Git commit

### PR DESCRIPTION
**Pull request by tengyifei**
Okay this one works but is definitely not as clean as I wanted. And there are way too many levels of nesting. Thoughts?

### CODE REVIEW

**Comment by mdiep** [comment]
Would you mind rebasing or merging in master? 🙏

File: SwiftGit2/Objects.swift [hunk 1]
```diff
@@ -37,13 +38,35 @@ public struct Signature {
...
 	public init(_ signature: git_signature) {
 		name = String(validatingUTF8: signature.name)!
 		email = String(validatingUTF8: signature.email)!
 		time = Date(timeIntervalSince1970: TimeInterval(signature.when.time))
 		timeZone = TimeZone(secondsFromGMT: 60 * Int(signature.when.offset))!
 	}
+
+	/// Return an unsafe pointer to the `git_signature` struct.
+	/// Caller is responsible for freeing it with `git_signature_free`.
+	var unsafeSignature: Result<UnsafeMutablePointer<git_signature>, NSError> {
```
Comment (mdiep) [inline SwiftGit2/Objects.swift]: It think we should probably make this a `makeX` function. That'd be more in line with the Swift API guidelines. `makeUnsafeSignature()`. Or maybe even `makeGitSignature()`.

Comment (tengyifei) [reply SwiftGit2/Objects.swift]: `makeUnsafeSignature()` sounds more consistent with our current `unsafe`-prefix stuff

**Comment by tengyifei** [comment]
@mdiep done

**Comment by mdiep** [comment]
Looks good apart from [this](https://github.com/SwiftGit2/SwiftGit2/pull/128#discussion_r186065309).

**Comment by tengyifei** [comment]
Thanks for taking the time to review!

File: SwiftGit2/Repository.swift [hunk 2]
```diff
@@ -593,23 +593,109 @@ final public class Repository {
...
+
+			// libgit2 expects a C-like array of parent git_commit pointer
+			var parentGitCommits: [OpaquePointer?] = []
+			for parentCommit in parents {
+				var parent: OpaquePointer? = nil
+				var oid = parentCommit.oid.oid
+				let lookupResult = git_commit_lookup(&parent, self.pointer, &oid)
+				guard lookupResult == GIT_OK.rawValue else {
+					let err = NSError(gitError: lookupResult, pointOfFailure: "git_commit_lookup")
+					return .failure(err)
```
[the remark this answers is not in the document]
Comment (tengyifei) [reply SwiftGit2/Repository.swift]: Doesn't a `defer` free the commit as soon as it goes out of scope i.e. the for loop?

[the remark this answers is not in the document]
Comment (tengyifei) [reply SwiftGit2/Repository.swift]: nvm I think I found a way

**Pull request merged**